.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on:
```

Comment:
i think we should run this on pr. look at:

https://github.com/RHVoice/RHVoice/blob/770546edd05dc237f098c9f3867e1702bd521806/.github/workflows/codeql-analysis.yml#L3-L8